engine/orchestration/src/org/apache/cloudstack/engine/orchestration/VolumeOrchestrator.java

```diff
@@ -1369,6 +1369,13 @@ public void prepare(VirtualMachineProfile vm, DeployDestination dest) throws Sto
             if (task.type == VolumeTaskType.NOP) {
                 pool = (StoragePool)dataStoreMgr.getDataStore(task.pool.getId(), DataStoreRole.Primary);
                 vol = task.volume;
+                if (pool.isManaged()) {
```

Comment:
@syed I am wondering if we should instead call revokeAccess on the volume for the old cluster and grantAccess on the volume for the new cluster. That way the storage plug-in doesn't need to have any new, special logic around automatically removing the volume from an existing VAG when adding it to a new VAG. It also provides a level of symmetry in CloudStack's core code around grant/revokeAccess (the core code never gets into a grantAccess + grantAccess scenario - it's always grantAccess + revokeAccess + grantAccess + revokeAccess, etc.).

What do you think?

Reply:
I think you are right @mike-tutkowski It definitely simplifies things.